### PR TITLE
dsq: init at 0.9.0

### PR DESCRIPTION
**Pull request by liff**
###### Description of changes

Add a package for [dsq](https://github.com/multiprocessio/dsq), which is a “Commandline tool for running SQL queries against JSON, CSV, Excel, Parquet, and more.”

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by liff** [comment]
Marked as broken on x86_64-darwin because it looks like [this issue](https://github.com/NixOS/nixpkgs/pull/157416#issuecomment-1045990033).

File: pkgs/tools/misc/dsq/default.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        cat <<EOF > "$out/expected.txt"
+        +-------+--------+
+        | first | second |
+        +-------+--------+
+        |     1 | a      |
+        |     2 | b      |
+        +-------+--------+
+        EOF
+        ${dsq}/bin/dsq --pretty "$out/input.csv" 'select first, second from {}' > "$out/actual.txt"
+        ${diffutils}/bin/diff "$out/expected.txt" "$out/actual.txt"
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/dsq/default.nix:44]: Please put this into nativeBuildInputs.

File: pkgs/tools/misc/dsq/default.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+, stdenv
+, fetchFromGitHub
+, buildGo118Module
+, runCommand
+, nix-update-script
+, dsq
+, diffutils
+}:
+
+buildGo118Module rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/dsq/default.nix:11]: Ideally this should be done in top-level.

Comment (liff) [reply pkgs/tools/misc/dsq/default.nix]: Moved, thanks.

**Pull request merged**